citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java

```diff
@@ -107,27 +169,118 @@ public IAuthorityEntry find(String id) {
+                .findPersonsByCitationGroupAndNameLike((ICitationGroup) group.get(), firstName, lastName, paging);
+        Set<IAuthorityEntry> entries = new HashSet<>();
+        persons.forEach(p -> {
+            Optional<AuthorityEntry> optional = entryRepository.findById(p.getLocalAuthorityId());
+            if (optional.isPresent()) {
+                entries.add(optional.get());
+            }
+        });
+        return entries;
+    }
+
+    @Override
+    public Set<IAuthorityEntry> findByNameInDataset(IUser user, String firstName, String lastName,
+            String citationGroupId, int page, int pageSize) throws GroupDoesNotExistException {
+        List<String> uriList = authorityRepository
+                .findByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(), firstName, lastName)
+                .stream().map(IAuthorityEntry::getUri).collect(Collectors.toList());
+
+        if (uriList != null && uriList.size() > 0) {
+            return this.findByNameInDataset(firstName, lastName, citationGroupId, uriList, page, pageSize);
+        } else {
+            return this.findByNameInDataset(firstName, lastName, citationGroupId, page, pageSize);
+        }
+    }
+
     @Override
     public boolean deleteAuthority(String id) {
         entryRepository.deleteById(id);
         return true;
     }
-    
+
     @Override
     public List<IAuthorityEntry> getAll(IUser user) {
         return entryRepository.findByUsernameOrderByName(user.getUsername());
     }
-    
+
     @Override
     public IAuthorityEntry create(IAuthorityEntry entry, IUser user) {
         entry.setUsername(user.getUsername());
         entry.setCreatedOn(OffsetDateTime.now());
         return save(entry);
     }
-    
+
     @Override
     public IAuthorityEntry save(IAuthorityEntry entry) {
-        return (IAuthorityEntry) entryRepository.save((AuthorityEntry)entry);
+        return (IAuthorityEntry) entryRepository.save((AuthorityEntry) entry);
+    }
+
+    @Override
+    public int getTotalUserAuthoritiesPages(IUser user, String firstName, String lastName, int pageSize) {
+
+        return (int) Math.ceil(new Float(authorityRepository
+                .countByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(), firstName, lastName))
+                / pageSize);
     }
+
+    @Override
+    public int getTotalDatasetAuthoritiesPages(String citationGroupId, String firstName, String lastName,
+            int pageSize) {
+        Optional<CitationGroup> group = groupRepository.findById(new Long(citationGroupId));
+
+        return (int) Math.ceil(new Float(personAuthorityRepository
+                .countByPersonsByCitationGroupAndNameLike((ICitationGroup) group.get(), firstName, lastName))
+                / pageSize);
+    }
+
+    public String getImporterSearchString(String source, String firstName, String lastName) {
+        if (source.equals(AuthorityImporter.CONCEPTPOWER)) {
```

Comment:
since this will be depending on the source that's being queries, this should be a method of the importer not of this class.